### PR TITLE
Test invite response for valid and invalid client ids

### LINKED ISSUE #285: Using an invalid client_id in user invitation url leads to a 500
## EXPECTED BEHAVIOR
If the user messes with the client_id field in the URL they will be taken to a 404 page instead of a 500

## ACTUAL BEHAVIOR
Taken to 500 if client_id field is messed with

## STEPS TO REPRODUCE
1. Invite a user
1. Copy the url in the email
1. Paste it into a browser and edit the client_id field
1. Press enter
1. Taken to a 500

## ADDITIONAL NOTES
Related to #274

### PR DESCRIPTION
**Pull request by Mistweaver**
Test creates a client and sends an invite for a new user.  An unmodified client_id returns a 200, an invalid/modified client_id will return a 404 instead of a 500.

Tests solution in #466 
Closes #285

### CODE REVIEW

File: api/test/application/api/email-verify-get.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    };
+
+    const res = await server.inject({
+      method: "POST",
+      url: "/api/invite",
+      credentials: {
+        scope: "admin",
+      },
+      payload,
+    });
```
Comment (Mistweaver) [inline api/test/application/api/email-verify-get.js:57]: Create user invite

File: api/test/application/api/email-verify-get.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+    const user = res.result;
+    const [{ token }] = await knex("SIP_email_token").where({
+      user_id: user.id,
+    });
```
Comment (Mistweaver) [inline api/test/application/api/email-verify-get.js:62]: Retrieve token from DB

File: api/test/application/api/email-verify-get.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+    const query = {
+      client_id,
+      redirect_uri,
+      response_type,
+      scope,
+      token,
+    };
```
Comment (Mistweaver) [inline api/test/application/api/email-verify-get.js:70]: Build query to stringily

File: api/test/application/api/email-verify-get.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    const verificationResponse = await server.inject({
+      method: "GET",
+      url: `/user/accept-invite?${querystring.stringify(query)}`.replace(
+        " ",
+        "%20"
+      ),
+    });
+
+    expect(verificationResponse.statusCode).to.equal(200);
+  });
```
Comment (Mistweaver) [inline api/test/application/api/email-verify-get.js:81]: Visit URL build from query, expect good things

File: api/test/application/api/email-verify-get.js [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+      payload,
+    });
+
+    const user = res.result;
+    const [{ token }] = await knex("SIP_email_token").where({
+      user_id: user.id,
+    });
+
+    const query = {
+      client_id: client_id + "junk",
```
Comment (Mistweaver) [inline api/test/application/api/email-verify-get.js:114]: Invalidate the client id

File: api/test/application/api/email-verify-get.js [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+    };
+
+    const verificationResponse = await server.inject({
+      method: "GET",
+      url: `/user/accept-invite?${querystring.stringify(query)}`.replace(
+        " ",
+        "%20"
+      ),
+    });
+    expect(verificationResponse.statusCode).to.equal(404);
```
Comment (Mistweaver) [inline api/test/application/api/email-verify-get.js:128]: Expect bad things

File: api/test/application/api/email-verify-get.js (hunk 2 above)
Comment (spruce-bruce) [reply api/test/application/api/email-verify-get.js:62]: How much of a pain would it be to retrieve the token from the email message sent to the mocked email service?

**Review by spruce-bruce — COMMENTED** [review]
Got a Q

**Review by spruce-bruce — APPROVED** [review]

**Pull request merged**